rt_gene/src/rt_gene/extract_landmarks_method.py

```diff
@@ -394,7 +397,7 @@ def publish_pose(self, timestamp, nose_center_3d_tf, rot_head, subject_id):
         self.tf_broadcaster.sendTransform(nose_center_3d_tf,
                                           rot_head,
                                           timestamp,
-                                          self.tf_prefix + "/head_pose_estimated" + str(subject_id),
+                                          self.tf_prefix + "/head_pose_estimated_" + str(subject_id[:5]),
```

Comment:
What's the reasoning behind the [:5]?

Reply:
I think for compatibility with downstream code, it would be nicer to just use `subject_id[-1]` and remove the `_`. What do you think?

/cc @ngageorange

Reply:
Yes please - the advertised topics and tf frames should stay consistent. I would suggest that the topic (& tf frames) name changes should only be changed for very good reasons that are debated and agreed upon rather than purely for aesthetic reasons.

Reply:
Hey @Tobias-Fischer and @ngageorange ,

The current id is generated using uuid, which is quite long. This is why I sliced the 5 first element of the id.

The old id system was simple, with integers:
If we have 3 subjects, the ids are [0, 1, 2], but if [1] leaves the scene, then we have [0, 1] where [2] becomes [1]. So we can lose the tracking...

I propose you to get back to a better solution:
First we remove the  `_` to stay consistent, effectively. 
Then we just use a unique counter:
If we have 3 subjects (the 3 first), the ids are [0, 1, 2], if [1] leaves the scene, then we have [0, 2], and if someone else (even the older [1]) appears we have [0, 2, 3]... and so on.

With this solution:
- we stay consistent and close to the old id system
- we have no problem with the size of the id 
- we do not lose the tracking anymore

I look forward to your feedback.

Best,
Kévin

Reply:
@Twarz I completely agree with your solutions.
I've created a new pull-request that addresses those points.

The other changes in this pull-request are excellent.

Reply:
Agreed. So once the changes to this line of code are removed, we can merge this PR, right?

Reply:
#22 is already merged.

Reply:
Done!

Reply:
Cool, thanks! @ngageorange are you happy, too? Then I'll merge.

Reply:
Very happy.